Source/DafnyCore/JsonDiagnostics.cs

```diff
@@ -61,9 +61,10 @@ record DiagnosticMessageData(MessageSource source, ErrorLevel level, Boogie.ITok
   }
 
   private static IEnumerable<JsonNode> SerializeInnerTokens(Boogie.IToken tok) {
-    while (tok is NestedToken ntok) {
-      tok = ntok.Inner;
-      yield return SerializeRelated(tok, null, "Related location");
+    while (tok is NestedToken nestedToken) {
+      tok = nestedToken.Inner;
+      var message = nestedToken.Message != null ? "Related location: " + nestedToken.Message : "Related location";
```

Comment:
Good job updating this! I don't think it is tested, is it?

Reply:
It is, in `Source/IntegrationTests/bin/Test/cli/diagnosticsFormats.dfy`